curation/data_sources/uta.py

```diff
@@ -0,0 +1,299 @@
+"""Module for UTA queries for genomic and transcript data."""
+from curation import UTA_DB_URL
+from six.moves.urllib import parse as urlparse
+import psycopg2
+import psycopg2.extras
+from typing import Dict, List, Optional, Tuple
+from os import environ
+import logging
+
+logger = logging.getLogger('fusion_backend')
+logger.setLevel(logging.DEBUG)
+
+
+class UTA:
+    """Class for accessing UTA database."""
+
+    def __init__(self, db_url=UTA_DB_URL, db_pwd=None) -> None:
+        """Initialize UTA class.
+
+        :param str db_url: UTA DB url
+        :param str db_pwd: UTA user uta_admin's password
+        """
+        self.db_url = self._update_db_url(db_pwd, db_url)
+        self.url = ParseResult(urlparse.urlparse(self.db_url))
+        self.schema = self.url.schema
+        self.args = self._get_conn_args()
+        self.conn = psycopg2.connect(**self.args)
+        self.conn.autocommit = True
+        self.cursor = \
+            self.conn.cursor(cursor_factory=psycopg2.extras.DictCursor)
+
+    @staticmethod
+    def _update_db_url(db_pwd, db_url) -> Optional[str]:
+        """Return new db_url containing password.
+
+        :param str db_pwd: uta_admin's user password
+        :param str db_url: PostgreSQL db url
+        :return: PostgreSQL db url with password included
+        """
+        if not db_pwd and 'UTA_PASSWORD' not in environ:
+            raise Exception('Environment variable UTA_PASSWORD '
+                            'or `db_pwd` param must be set')
+        else:
+            uta_password_in_environ = 'UTA_PASSWORD' in environ
+            db_url = db_url.split('@')
+            if uta_password_in_environ and db_pwd:
+                if db_pwd != environ['UTA_PASSWORD']:
+                    raise Exception('If both environment variable UTA_'
+                                    'PASSWORD and param db_pwd is set, '
+                                    'they must both be the same')
+            else:
+                if uta_password_in_environ and not db_pwd:
+                    db_pwd = environ['UTA_PASSWORD']
+            return f"{db_url[0]}:{db_pwd}@{db_url[1]}"
+
+    def _get_conn_args(self) -> Dict:
+        """Return connection arguments.
+
+        :return: A dictionary containing db credentials
+        """
+        return dict(
+            host=self.url.hostname,
+            port=self.url.port,
+            database=self.url.database,
+            user=self.url.username,
+            password=self.url.password,
+            application_name='gene-fusions',
+        )
+
+    def get_genomic_coords(self, tx_ac, start_exon, end_exon, start_exon_offset=0,
+                           end_exon_offset=0, gene=None) -> Dict:
```

Comment:
If you have a sec this week, could you explain how the offset values are used? I think I have it sort of clarified from reading this but I just want to double-check that I understand